stingray/fourier.py

```diff
@@ -1922,3 +1925,207 @@ def avg_cs_from_events(
+
+    fullspec : bool, optional, default: False
+        Return LSFT values for full frequency array (True) or just positive frequencies (False).
+
+    oversampling : float, optional, default: 5
+        Interpolation Oversampling Factor
+
+    Returns
+    -------
+    ft_res : numpy.ndarray
+        An array of Fourier transformed data.
+    """
+    y_ = copy.deepcopy(y) - np.mean(y)
+    freqs = freqs[freqs >= 0]
+    # Constants initialization
+    sum_xx = np.sum(y_)
+    num_xt = len(y_)
+    num_ww = len(freqs)
+
+    # Arrays initialization
+    ft_real = ft_imag = np.zeros((num_ww))
+    f0, df, N = freqs[0], freqs[1] - freqs[0], len(freqs)
+
+    # Sum (y_i * cos(wt - wtau))
+    Sh, Ch = trig_sum(t, y_, df, N, f0, oversampling=oversampling)
+
+    # Angular Frequency
+    w = freqs * 2 * np.pi
+
+    # Summation of cos(2wt) and sin(2wt)
+    csum, ssum = trig_sum(
+        t, np.ones_like(y_) / len(y_), df, N, f0, freq_factor=2, oversampling=oversampling
+    )
+
+    wtau = 0.5 * np.arctan2(ssum, csum)
+
+    S2, C2 = trig_sum(
+        t,
+        np.ones_like(y_),
+        df,
+        N,
+        f0,
+        freq_factor=2,
+        oversampling=oversampling,
+    )
+
+    const = np.sqrt(0.5) * np.sqrt(num_ww)
+
+    coswtau = np.cos(wtau)
+    sinwtau = np.sin(wtau)
+
+    sumr = Ch * coswtau + Sh * sinwtau
+    sumi = Sh * coswtau - Ch * sinwtau
+
+    cos2wtau = np.cos(2 * wtau)
+    sin2wtau = np.sin(2 * wtau)
+
+    scos2 = 0.5 * (N + C2 * cos2wtau + S2 * sin2wtau)
+    ssin2 = 0.5 * (N - C2 * cos2wtau - S2 * sin2wtau)
+
+    ft_real = const * sumr / np.sqrt(scos2)
+    ft_imag = const * np.sign(sign) * sumi / np.sqrt(ssin2)
+
+    ft_real[freqs == 0] = sum_xx / np.sqrt(num_xt)
+    ft_imag[freqs == 0] = 0
+
+    phase = wtau - w * t[0]
+
+    ft_res = np.complex128(ft_real + (1j * ft_imag)) * np.exp(-1j * phase)
+
+    return ft_res
+
+
+def lsft_slow(
+    y: npt.ArrayLike,
+    t: npt.ArrayLike,
+    freqs: npt.ArrayLike,
+    sign: Optional[int] = 1,
+) -> npt.NDArray:
+    """
+    Calculates the Lomb-Scargle Fourier transform of a light curve.
+    Only considers non-negative frequencies.
+    Subtracts mean from data as it is required for the working of the algorithm.
+
+    Parameters
+    ----------
+    y : a `:class:numpy.array` of floats
+        Observations to be transformed.
+
+    t : `:class:numpy.array` of floats
+        Times of the observations
+
+    freqs : numpy.ndarray
+        An array of frequencies at which the transform is sampled.
+
+    sign : int, optional, default: 1
+        The sign of the fourier transform. 1 implies positive sign and -1 implies negative sign.
+
+    fullspec : bool, optional, default: False
+        Return LSFT values for full frequency array (True) or just positive frequencies (False).
+
+    Returns
+    -------
+    ft_res : numpy.ndarray
+        An array of Fourier transformed data.
+    """
+    y_ = copy.deepcopy(y) - np.mean(y)
+    freqs = freqs[freqs >= 0]
```

Comment:
in case the user gives a list of frequencies, this should not fail. Better to wrap `freqs` with `asarray`